applets/clock/clock.c

```diff
@@ -888,105 +893,181 @@ create_calendar (ClockData *cd)
+        }
+#endif
 
-                break;
-        case MATE_PANEL_APPLET_ORIENT_UP:
-                y -= h;
-                if ((x + w) > monitor.x + monitor.width)
-                        x -= (x + w) - (monitor.x + monitor.width);
+    /*Only build wayland support when building in process*/
+#if defined(CLOCK_INPROCESS) && defined(HAVE_WAYLAND) && defined(GDK_WINDOWING_WAYLAND)
+        if (GDK_IS_WAYLAND_DISPLAY (gdk_display_get_default ()))
+        {
+                GtkWindow *window;
+                GdkWindow *panelwin;
+                GtkWidget *toplevel;
+                int  x, y, w, h, panel_w, panel_h;
+
+                /*Get the calendar window dimensions*/
+                window = (GTK_WINDOW (cd->calendar_popup));
+                gtk_window_get_size (window, &w, &h);
+                /*Find the position of the applet*/
+                gdk_window_get_origin (gtk_widget_get_window (cd->panel_button),
+                                                                      &x, &y);
+
+                /*Get the panel dimensions*/
+                toplevel = gtk_widget_get_toplevel (cd->applet);
+                panelwin = gtk_widget_get_window (toplevel);
+                gdk_window_get_geometry (panelwin, NULL, NULL, &panel_w, &panel_h);
+
+                /*Set up GTK Layer Shell*/
+                gtk_layer_init_for_window (window);
+                gtk_layer_set_layer (window, GTK_LAYER_SHELL_LAYER_TOP);
+
+                switch (cd->orient) {
+                case MATE_PANEL_APPLET_ORIENT_RIGHT:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+                        if (y < (panel_h -h))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_TOP, y);
+                        }
+                        else
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
 
-                gravity = GDK_GRAVITY_SOUTH_WEST;
+                        break;
+                case MATE_PANEL_APPLET_ORIENT_LEFT:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
+                        if (y < (panel_h - h))
+                        {
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                                gtk_layer_set_margin (window, GTK_LAYER_SHELL_EDGE_TOP, y);
+                        }
+                        else
+                                gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
 
-                break;
-        }
+                        break;
+                case MATE_PANEL_APPLET_ORIENT_DOWN:
+                        gtk_layer_set_anchor (window, GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+                        if (x < (panel_w -w))
```

Comment:
style: space around the `-` binary operator

Reply:
Found and fixed